src/shogun/mathematics/linalg/LinalgBackendBase.h

```diff
@@ -58,7 +67,14 @@ class LinalgBackendBase
 	BACKEND_GENERIC_DOT(float64_t);
 	BACKEND_GENERIC_DOT(int32_t);
 
-	/** Generic transfer_to_gpu macro */
+	/**
```

Comment:
Actually, here is another thing.
Can you add an additional abstract base class for LinalgBackendGPUBase. And put the to_cpu from_gpu methods there and make them purely virtual? This way we enforce that GPU backends implement them
Furthermore, you can have different pointer types for gpu_backend and cpu_backend.
Sorry should have said that before.